mlflow/pytorch/__init__.py

```diff
@@ -297,7 +297,11 @@ def save_model(pytorch_model, path, conda_env=None, mlflow_model=None, code_path
         f.write(pickle_module.__name__)
     # Save pytorch model
     model_path = os.path.join(model_data_path, _SERIALIZED_TORCH_MODEL_FILE_NAME)
-    torch.save(pytorch_model, model_path, pickle_module=pickle_module, **kwargs)
+    if isinstance(pytorch_model, torch.jit.ScriptModule):
```

Comment:
BTW, we'd actually decided in https://github.com/mlflow/mlflow/issues/2263 that it made more sense to add torchscript model persistence support via a separate model flavor. There are two issues with having `mlflow.pytorch.log_model` support logging `torchscript` models:

1. At least currently, `mlflow.pytorch.load_model` cannot load back `torch.jit.ScriptModule` models (since we call `torch.load` to load back the model)
2. If we did add support for loading back `torch.jit.ScriptModule`s, downstream user code would have to type-check whether the loaded model was an instance of `torch.jit.ScriptModule` or `torch.nn.Module` after calling `loaded_model = mlflow.pytorch.load_model(...)`

Thus we'd discussed that it'd actually be cleaner to have separate `mlflow.torchscript.log_model/load_model` APIs. Let me know if that makes sense, or if I'm missing something!

Reply:
@chauhang Can you please confirm if we can add a separate flavor for torch scripted models. We remember discussing the same during the call, where you had suggested us not to create a separate flavor.

Reply:
If there's a good way to add unified pytorch/torchscript support that'd be awesome, but at least I'm not aware of an approach that satisfies the issues mentioned above^. Note that even with a separate torchscript flavor we could consider having `mlflow.pytorch.log_model` additionally generate the torchscript flavor, i.e. call `torch.jit.trace` on the input model, if it's not too expensive to do so.

Reply:
We discussed this on the call and it is resolved. We are adding examples for the pyfunc as well